### PR TITLE
Skills

### PR DESCRIPTION
**Pull request by Loerad**
## Describe your changes

This PR adds the skills dash, decoy and vanish as well as the menu for clicking the skills before the game starts.

image
This is a mockup of the skills screen with the images I made in Krita by tracing some stock images of ducks.

image
This is what the new UI looks like in game and the decoy skill in action

I have also added a command to change the currently active skill and fixed the togglespawn command feedback to be helpful now. The setskill command does not change the image of the skill when the game is running. I don't plan on fixing that.

There is a "bug" that sends null reference exceptions when you move before shooting when you play the game. It appears to not have any effect on gameplay and only seems to be slightly annoying. #96 for more info

I did dabble into some of the enemy scripts so hopefully there isn't too many merge conflicts. Let me know what you think!

## Issue number and link

#10 
#83 
#87 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Loerad** [comment]
Pr is ready to review

**Review by AlexReidNZ — APPROVED** [review]
Skills function well, are fun to use and have clear visuals. Ready for merge. 

We should maybe consider starting cooldowns when the skill ends rather than when it starts (matters more for vanish as the duration is quite long).

**Review by PBWiseman — CHANGES_REQUESTED** [review]
Very good feature. The only bug I could find is that when there is a decoy down and you walk into a bullet not fired at you the bullet just pushes you around. The dog bites will damage you but I assume this is intentional.

On the balance end I agree with the comment about the cooldown starting at the end of the duration making more sense. 

For the dash did we end up deciding that it shouldn't have any invulnerability frames? If so the wording over the description should be changed to remove the word dodge.

Couple other things that might be personal preference. I think that movemode for the dashing feels a lot better but we should consider having the stand still dash use mouse mode. Always dashing upwards for that feels odd.

Vanish and Decoy both feel good. They feel quite strong, but I don't think that's an issue. The only other thing is considering if there should be something to identify which key to press next to the button or if that just being in settings is fine?

**Comment by Loerad** [comment]
Fixing, the cooldown starting after the duration has ended. I have made it so dashing with no input does nothing and does not consume the dash. I will add some I-frames to the dash. I will have a look at fixing the decoy bullet issue, I may make it destroy the bullets it touches

**Comment by Loerad** [comment]
Ready for re-review

File: Assets/Scripts/EnemyScripts/Bosses/BossBullet.cs [hunk 1]
```diff
@@ -75,9 +75,18 @@ void Update()
...
-            player.GetComponent<PlayerStats>().ReceiveDamage(bulletDamage);
+            if (SkillEffects.Instance.decoyActive)
+            {
+                player = GameObject.FindGameObjectWithTag("Player");
+                player.GetComponent<PlayerStats>().ReceiveDamage(bulletDamage);
+                player = GameObject.FindGameObjectWithTag("Decoy");
+            }
+            else
+            {
+                player.GetComponent<PlayerStats>().ReceiveDamage(bulletDamage);
```
Comment (Loerad) [inline Assets/Scripts/EnemyScripts/Bosses/BossBullet.cs:88]: Change to other.Get...

**Review by PBWiseman — APPROVED** [review]
Issues fixed. Happy for this to be merged

**Review by kblack89 — APPROVED** [review]
After playtesting this out multiple times it works well with no apparent issues. The code looks good and makes sense and is nice and clean.

**Pull request merged**